src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,115 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int INITIAL_CAPACITY = 10;
+    private T[] elements;
+    private int size = 0;
+
+    public ArrayList() {
+        this.elements = (T[]) new Object[INITIAL_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        resizeArrayIfFull();
+        elements[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        checkIndexForAdd(index);
+        resizeArrayIfFull();
+        if (elements[index] == null) {
+            elements[index] = value;
+        } else {
+            System.arraycopy(elements, index, elements, index + 1, size - index);
+            elements[index] = value;
+        }
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        resizeArrayIfFull();
+        for (int i = 0; i < list.size(); i++) {
+            elements[size] = list.get(i);
+            size++;
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkElementExistsByIndex(index);
+        return elements[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        checkElementExistsByIndex(index);
+        elements[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkElementExistsByIndex(index);
+        T removed = elements[index];
+        System.arraycopy(elements, index + 1, elements, index, size - index - 1);
+        size--;
+        return removed;
     }
 
     @Override
     public T remove(T element) {
-        return null;
+        int index = findByValue(element);
+        checkElementExistsByIndex(index);
```

Comment:
```suggestion
```
no need to check index, if there is value present - index is ok
if there is no value exception would be thrown